### PR TITLE
ci: Relocate insight execution from a workflow_run to a post-lint step

### PR DESCRIPTION
**Pull request by bjhargrave**
We were not getting the desired behavior from a workflow_run.

### CODE REVIEW

**Comment by bjhargrave** [comment]
cc @nitin-ramchandani

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -78,3 +78,10 @@ jobs:
           TAXONOMY_FOLDERS: >-
             compositional_skills
             knowledge
+
+      - name: "Invoke insight handler"
+        if: ${{ (steps.changed-files.outputs.any_changed == 'true') && (github.event_name == 'pull_request') && (github.repository == 'instructlab/taxonomy') }}
+        run: |
+          echo "${{ toJSON(github.event) }}"
+          curl -X 'POST' "https://pr-analysis-handler.1fuhf5gskmng.us-east.codeengine.appdomain.cloud/analyze-pr?pr_number=${{ github.event.number }}&owner=${{ github.repository_owner }}&repo=${{ github.event.repository.name }}" -H 'accept: application/json' -H 'Authorization: Bearer ${{ secrets.COC_ANALYSIS_TOKEN }}' -d '' &
```
Comment (jjasghar) [inline .github/workflows/lint.yml:86]: Where are we actually sending the json, -d is ' '?

Comment (bjhargrave) [reply .github/workflows/lint.yml:86]: @nitin-ramchandani Can you answer? I assume the endpoint only cares about the parameters.

Comment (nitin-ramchandani) [reply .github/workflows/lint.yml:86]: @bjhargrave yes just parameters are good enough, I think empty payload argument got carried over from test workflow

Comment (bjhargrave) [reply .github/workflows/lint.yml:86]: @jjasghar Are you ok with this information to approve?

Comment (jjasghar) [reply .github/workflows/lint.yml:86]: Ah ok. Just checking.

**Review by jjasghar — APPROVED** [review]

**Pull request merged**